src/ring_theory/localization.lean

```diff
@@ -127,7 +127,11 @@ variables (f : localization_map M S)
 
 /-- We define a copy of the localization map `f`'s codomain `S` carrying the data of `f` so that
 instances on `S` induced by `f` can 'know` the map needed to induce the instance. -/
-@[reducible, nolint unused_arguments] def codomain (f : localization_map M S) := S
+@[nolint unused_arguments has_inhabited_instance]
+def codomain (f : localization_map M S) := S
+
+instance : comm_ring f.codomain := _inst_2
```

Comment:
```suggestion
instance : comm_ring f.codomain := by apply_instance
```
`_inst_2` doesn't seem like a very stable name. If someone adds a variable we might suddenly need `_inst_3` here.

Reply:
fair point! ```by apply_instance``` doesn't work but I've just found out you can use ```by assumption``` for instances... awesome